base/ntuple.jl

```diff
@@ -42,6 +42,23 @@ ntuple(f, ::Val{1}) = (@_inline_meta; (f(1),))
 ntuple(f, ::Val{2}) = (@_inline_meta; (f(1), f(2)))
 ntuple(f, ::Val{3}) = (@_inline_meta; (f(1), f(2), f(3)))
 
+"""
+    ntuple(f::Function, ::Val{N})
+
+Create a tuple of length `N`, computing each element as `f(i)`,
+where `i` is the index of the element. By taking a `Val(N)` 
+argument, it is possible that this version of ntuple may
+generate more efficient code than the version taking the 
+length as an integer. But `ntuple(fun, N)` is preferable
+to `ntuple(fun, Val(N))` in cases where `N` cannot be
+determined at compile time.
```

Comment:
I think it is better to use `f` instead of `fun` to be consistent with the method signature.

Reply:
Agreed. Yes I'll change this.